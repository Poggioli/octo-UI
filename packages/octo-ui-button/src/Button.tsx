import { forwardRef } from "@octo-ui/core";
import { StyledButton } from "./styles";
import { ButtonProps } from "./types";

const Button = forwardRef<typeof StyledButton, ButtonProps>(
  ({ ...props }, forwardedRef) => {
    return <StyledButton {...props} ref={forwardedRef} />;
  }
);
Button.toString = () => `.${StyledButton.className}`;

export { Button };
export { StyledButton };
