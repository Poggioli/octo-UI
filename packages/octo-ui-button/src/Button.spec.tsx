import { render } from "@testing-library/react";
import { Button } from "./Button";

describe("<Button />", () => {
  it(`GIVEN a <Button />
      WHEN render is called
      THEN should render a <Button />`, () => {
    const rendered = render(<Button>Button</Button>);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Button />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Button.toString()).toBe(".octo-UI-c-fYBmDf");
  });
});
