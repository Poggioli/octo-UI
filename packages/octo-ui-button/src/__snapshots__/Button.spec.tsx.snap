// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> GIVEN a <Button />
      WHEN render is called
      THEN should render a <Button /> 1`] = `
<div>
  <button
    class="octo-UI-c-fYBmDf octo-UI-c-fYBmDf-gHVRac-color-unstyled octo-UI-c-fYBmDf-cYmrsO-variant-solid octo-UI-c-fYBmDf-flTkwI-corners-soft octo-UI-c-fYBmDf-JxRTl-size-standard"
  >
    Button
  </button>
</div>
`;
